core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -1086,11 +1086,19 @@ private[spark] class TaskSchedulerImpl(
     case ExecutorKilled =>
       logInfo(s"Executor $executorId on $hostPort killed by driver.")
     case _: ExecutorDecommission =>
-      logInfo(s"Executor $executorId on $hostPort is decommissioned.")
+      logInfo(s"Executor $executorId on $hostPort is decommissioned after " +
+        s"${getDecommissionDuration(executorId)}.")
     case _ =>
       logError(s"Lost executor $executorId on $hostPort: $reason")
   }
 
+  // return decommission duration in string or "unknown time" if decommission startTime not exists
+  private def getDecommissionDuration(executorId: String): String = {
+    executorsPendingDecommission.get(executorId)
+      .map(s => Utils.msDurationToString(clock.getTimeMillis() - s.startTime))
+      .getOrElse("unknown time")
```

Comment:
So, the log message goes with `... is decommissioned after unknown time`?

Reply:
This could give a not-good impression to the users because it's interpreted as a Spark bug by the users. Can we give more informative message to avoid that? Or, can we keep the original message in this case?

For example,
- When we know the time, `Executor $executorId on $hostPort is decommissioned after ...`
- When we don't know, `Executor $executorId on $hostPort is decommissioned.` (the same message like before)